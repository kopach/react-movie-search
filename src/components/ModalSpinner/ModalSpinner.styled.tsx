import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCompactDisc } from '@fortawesome/free-solid-svg-icons'

// TODO
export const matrixColor = 'rgba(122, 229, 114, 0.87)'

export const Spinner = styled(FontAwesomeIcon).attrs({
  icon: faCompactDisc,
  color: matrixColor,
  className: 'fa-spin fa-10x ',
})``

export const ContainerHorizontalCenter = styled.div.attrs({
  className: 'text-center',
})`
  margin-left: auto;
  margin-right: auto;
  position: relative;
`

export const ContainerFullScreen = styled.div.attrs({
  className: 'w-100 h-100',
})`
  position: fixed;
`
