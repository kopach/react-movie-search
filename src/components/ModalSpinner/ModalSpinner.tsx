import React from 'react'
import { ContainerHorizontalCenter, ContainerFullScreen, Spinner } from './ModalSpinner.styled'

export const ModalSpinner: React.FC = (): JSX.Element => {
  return (
    <ContainerHorizontalCenter>
      <ContainerFullScreen>
        <div className="row h-100">
          <div className="col-sm-12 my-auto">
            <Spinner />
          </div>
        </div>
      </ContainerFullScreen>
    </ContainerHorizontalCenter>
  )
}
