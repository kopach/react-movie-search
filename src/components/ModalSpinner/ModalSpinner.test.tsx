import React from 'react'
import { shallow } from 'enzyme'
import { ModalSpinner } from './ModalSpinner'

describe('<ModalSpinner />', () => {
  let component: ReturnType<typeof shallow>

  beforeEach(() => {
    component = shallow(<ModalSpinner />)
  })

  it('should mount', () => {
    expect(component.length).toBe(1)
  })

  it('renders as expected', () => {
    expect(component).toMatchSnapshot()
  })
})
