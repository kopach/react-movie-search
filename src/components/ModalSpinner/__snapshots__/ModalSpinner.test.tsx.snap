// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalSpinner /> renders as expected 1`] = `
<styled.div>
  <styled.div>
    <div
      className="row h-100"
    >
      <div
        className="col-sm-12 my-auto"
      >
        <Styled(FontAwesomeIcon) />
      </div>
    </div>
  </styled.div>
</styled.div>
`;
